src/main/java/brooklyn/entity/proxy/aws/ElbControllerImpl.java

```diff
@@ -173,175 +174,179 @@ protected void startLoadBalancer() {
-        String instanceProtocol = getRequiredConfig(INSTANCE_PROTOCOL);
-        String elbScheme = getConfig(LOAD_BALANCER_SCHEME);
-        String loadBalancerProtocol = getRequiredConfig(LOAD_BALANCER_PROTOCOL);
-        Collection<String> securityGroups = getConfig(LOAD_BALANCER_SECURITY_GROUPS);
-        Collection<String> subnets = getConfig(LOAD_BALANCER_SUBNETS);
-        String sslCertificateId = getConfig(SSL_CERTIFICATE_ID);
-        Set<String> availabilityZoneNames = getAvailabilityZones(loc);
-        Boolean healthCheckEnabled = getConfig(HEALTH_CHECK_ENABLED);
-        Integer healthCheckInterval = getConfig(HEALTH_CHECK_INTERVAL);
-        Integer healthCheckTimeout = getConfig(HEALTH_CHECK_TIMEOUT);
-        Integer healthCheckHealthyThreshold = getConfig(HEALTH_CHECK_HEALTHY_THRESHOLD);
-        Integer healthCheckUnhealthyThreshold = getConfig(HEALTH_CHECK_UNHEALTHY_THRESHOLD);
+        Set<String> availabilityZoneNames = (getConfig(AVAILABILITY_ZONES) != null) ? ImmutableSet.copyOf(getConfig(AVAILABILITY_ZONES)) : ImmutableSet.<String>of();
+        Set<String> subnets = (getConfig(LOAD_BALANCER_SUBNETS) != null) ? ImmutableSet.copyOf(getConfig(LOAD_BALANCER_SUBNETS)) : ImmutableSet.<String>of();
 
         Group serverPool = getConfig(SERVER_POOL);
         LOG.debug("Creating new ELB '" + elbName + "', for server-pool " + serverPool);
 
         String identity = loc.getIdentity();
         String credential = loc.getCredential();
-        LoadBalancerServiceContext loadBalancerServiceContext = null;
+
         try {
             loadBalancerServiceContext = newLoadBalancerServiceContext(identity, credential);
             // TODO would be cool to use the loadbalancer abstraction but it doesn't like an empty collection of nodes
-            ELBApi elbApi = api(loadBalancerServiceContext);
-            String elbDnsName = elbApi.getLoadBalancerApi()
-                    .createListeningInAvailabilityZones(
-                            elbName,
-                            Listener.builder()
-                                    .protocol(Protocol.HTTP)
-                                    .instancePort(instancePort)
-                                    .port(loadBalancerPort)
-                                    .build(),
-                            availabilityZoneNames);
+            String elbDnsName = "";
+            Listener listener = buildListener();
+//          create vanilla load balancer
+            if(availabilityZoneNames != null && availabilityZoneNames.size() > 0 ){
+                elbDnsName = getELBApi().getLoadBalancerApi().createLoadBalancerInAvailabilityZones(elbName, ImmutableSet.<Listener>of(listener), availabilityZoneNames);
+            }
+            else if(subnets != null && !subnets.isEmpty()) {
+                elbDnsName = getELBApi().getLoadBalancerApi().createLoadBalancerInSubnets(elbName, ImmutableSet.<Listener>of(listener), subnets);
+            }
```

Comment:
what is the correct exception to throw for config etc? and set it in the service problems?